File: my-app/src/app/models/Order.ts
export interface Order {
    id:              number;
    buyerId:         string;
    shippingAddress: ShippingAddress;
    orderDate:       string; //เปลี่ยนจาก Date เป็น string เพราะส่งมาเป็น json string 
    orderItems:      OrderItem[];
    subtotal:        number;
    deliveryFee:     number;
    orderStatus:     string;
    total:           number;
    voucher:         Voucher;
}

export interface OrderItem {
    productId:  number;
    name:       string;
    pictureUrl: string;
    price:      number;
    quantity:   number;
}

export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city:     string;
    state:    string;
    zip:      string;
    country:  string;
}

export interface Voucher {
    id:       number;
    name:     string;
    detail:   string;
    discount: number;
}

File: API/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities.OrderAggregate
{
    //ใบหลักการสั่งซื้อ one
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> OrderItems { get; set; }
        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string Voucher { get; set; }
        public string PaymentIntentId { get; set; }
        public long GetTotal()
        {
            if (Voucher == null)
            {
                return Subtotal + DeliveryFee;
            }
            else
            {
                return Subtotal + DeliveryFee;
            }

        }
    }

}